base/v0_5_exp/translate.go

```diff
@@ -218,6 +220,155 @@ func translateLink(from Link, options common.TranslateOptions) (to types.Link, t
+			tm.AddTranslation(c, path.New("json", fmt.Sprintf("sshAuthorizedKeys.%d", i)))
+			to.SSHAuthorizedKeys = append(to.SSHAuthorizedKeys, types.SSHAuthorizedKey(sshKey))
+		}
+	}
+
+	if from.SSHAuthorizedKeysLocal != nil && len(from.SSHAuthorizedKeysLocal) > 0 {
+		c = path.New("yaml", "ssh_authorized_keys_local")
+		tm.AddTranslation(c, path.New("json", "sshAuthorizedKeys"))
+		for _, sshKeyFile := range from.SSHAuthorizedKeysLocal {
+			if options.FilesDir == "" {
+				r.AddOnError(c, common.ErrNoFilesDir)
+				return
+			}
+
+			// calculate file path within FilesDir and check for
+			// path traversal
+			filePath := filepath.Join(options.FilesDir, sshKeyFile)
+			if err := baseutil.EnsurePathWithinFilesDir(filePath, options.FilesDir); err != nil {
+				r.AddOnError(c, err)
+				continue
+			}
+			contents, err := ioutil.ReadFile(filePath)
+			if err != nil {
+				r.AddOnError(c, err)
+				continue
+			}
+
+			sshKeyStrings := strings.TrimSpace(string(contents))
+			if len(contents) == 0 || len(sshKeyStrings) == 0 {
+				r.AddOnError(c, common.ErrFileEmpty)
+				continue
+			}
+			// offset for TranslationSets when both ssh_authorized_keys and ssh_authorized_keys_local are available
+			offset := len(to.SSHAuthorizedKeys)
+			for i, line := range strings.Split(sshKeyStrings, "\n") {
+				sshKey := strings.TrimSpace(line)
+				if len(sshKey) == 0 {
+					r.AddOnError(c, common.ErrSSHKeyEmpty)
+					continue
+				}
+				tm.AddTranslation(c, path.New("json", fmt.Sprintf("sshAuthorizedKeys.%d", i+offset)))
+				to.SSHAuthorizedKeys = append(to.SSHAuthorizedKeys, types.SSHAuthorizedKey(sshKey))
+			}
+		}
+	}
+
+	return
+}
+
+func translateSystemdUnits(from Unit, options common.TranslateOptions) (to types.Unit, tm translate.TranslationSet, r report.Report) {
+	tr := translate.NewTranslator("yaml", "json", options)
+	tr.AddCustomTranslator(translateSystemdDropInUnits)
+	tm, r = translate.Prefixed(tr, "name", &from.Name, &to.Name)
+	translate.MergeP(tr, tm, &r, "enabled", &from.Enabled, &to.Enabled)
+	translate.MergeP(tr, tm, &r, "mask", &from.Mask, &to.Mask)
+	translate.MergeP(tr, tm, &r, "dropins", &from.Dropins, &to.Dropins)
+
+	if from.Contents != nil && len(*from.Contents) > 0 {
+		translate.MergeP(tr, tm, &r, "contents", &from.Contents, &to.Contents)
+	}
+
+	if from.ContentsLocal != nil && len(*from.ContentsLocal) > 0 {
+		c := path.New("yaml", "contents")
+		if options.FilesDir == "" {
+			r.AddOnError(c, common.ErrNoFilesDir)
+			return
+		}
+
+		// calculate file path within FilesDir and check for
+		// path traversal
+		filePath := filepath.Join(options.FilesDir, *from.ContentsLocal)
+		if err := baseutil.EnsurePathWithinFilesDir(filePath, options.FilesDir); err != nil {
+			r.AddOnError(c, err)
+			return
+		}
+		contents, err := ioutil.ReadFile(filePath)
+		if err != nil {
+			r.AddOnError(c, err)
+			return
+		}
+		tm.AddTranslation(c, path.New("json", "contents"))
+		stringContents := string(contents)
+		to.Contents = &stringContents
```

Comment:
`ContentsLocal` and `Contents` are mutually exclusive, but `validate.go` doesn't check for that.

Also:

```suggestion
		to.Contents = util.StrToPtr(string(contents))
```